### PR TITLE
Server returns hard-coded comments data

### PR DESCRIPTION
**Pull request by TMazenge**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,28 +27,23 @@
 public class DataServlet extends HttpServlet {
 
   ArrayList<String> comments = new ArrayList<String>();
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: private access modifier?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,28 +27,23 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
     comments.add("What an amazing profile");
     comments.add("This is very awesome indeed.");
     comments.add("I want to travel to the Taj Mahal as well");
-    
-    String json = convertToJson(comments);
+
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("text/html;");
```
Comment (ankitaj224) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:38]: is the ; required?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -88,3 +88,22 @@ async function getDataAsyncAwait() {
   document.getElementById('data-container').innerText = data;
 }
 
+function getServerComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    // stats is an object, not a string, so we have to
```
Comment (ankitaj224) [inline portfolio/src/main/webapp/script.js:93]: Comment isn't clear. Whats stats?

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -88,3 +88,22 @@ async function getDataAsyncAwait() {
...
+    // reference its fields to create HTML content
+   
+    const commentsListElement = document.getElementById('comments-container');
```
Comment (ankitaj224) [inline portfolio/src/main/webapp/script.js:96]: commentsElement or commentListElement?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (TMazenge) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: ; is not required so i have removed it.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (TMazenge) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: Do I need to change the public to private?

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (TMazenge) [reply portfolio/src/main/webapp/script.js]: It was in reference to the walk-throughs so I removed it.

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (TMazenge) [reply portfolio/src/main/webapp/script.js]: Updated the variable name.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (ankitaj224) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:29]: private

File: portfolio/src/main/webapp/comments.html [hunk 5]
```diff
@@ -9,9 +9,11 @@
...
-      <button onclick="getDataAsyncAwait()">Data</button>
-      <div id="data-container"></div>
+       <p>Click here to see comments from server</p>
+      <button onclick="getServerComments()">Fetch server comments</button>
+      <ul id="comments-container"></ul>
     </div>
   </body>
 </html>
+
+
```
Comment (ankitaj224) [inline portfolio/src/main/webapp/comments.html:19]: newline at the eof

**Pull request merged**